C/plugins/north/OMF/omf.cpp

```diff
@@ -1161,9 +1162,17 @@ uint32_t OMF::sendToServer(const vector<Reading *>& readings,
 	bool legacyType = m_legacy;
 
 	// Create the class that deals with the linked data generation
-	OMFLinkedData linkedData(&m_containerSent, &m_assetSent, &m_linkSent, m_PIServerEndpoint);
+	OMFLinkedData linkedData(&m_linkedAssetState, m_PIServerEndpoint);
 	linkedData.setFormats(getFormatType(OMF_TYPE_FLOAT), getFormatType(OMF_TYPE_INTEGER));
 
+	for (vector<Reading *>::const_iterator elem = readings.begin();
+						    elem != readings.end();
+						    ++elem)
+	{
+		Reading *reading = *elem;
+		linkedData.buildLookup(*reading);
```

Comment:
Can we pass the whole readings vector to buildLookup() and let it iterate over the constituent readings in the vector? Please ignore this comment if individual readings are also to be passed to this function in certain cases.

Reply:
Sure, it will save a number of function call overheads